R/learner_stats_regr_glm.R

```diff
@@ -0,0 +1,116 @@
+#' @title Regression Generalized Linear Model Regression Learner
+#' @author salauer
+#' @name mlr_learners_regr.glm
+#'
+#' @template class_learner
+#' @templateVar id regr.glm
+#' @templateVar caller glm
+#'
+#'
+#' @template seealso_learner
+#' @template example
+#'
+#' @section Custom mlr3 defaults:
+#' - `type`
+#'   - Actual default: "link"
+#'   - Adjusted default: "response"
+#'   - Reason for change: Users rarely want predictions on the link scale
+#'   (for instance, on the log scale when using the Poisson distribution).
+#'   It would be much more reasonable for the default to be "response",
+#'   which puts predictions on the scale of the observed data,
+#'   and allow the user to change the type to "link" if they so desire.
+#' @export
+LearnerRegrGlm = R6Class("LearnerRegrGlm",
+  inherit = LearnerRegr,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamLgl$new("singular.ok", default = TRUE, tags = "train"),
+        ParamLgl$new("x", default = FALSE, tags = "train"),
+        ParamLgl$new("y", default = TRUE, tags = "train"),
+        ParamLgl$new("model", default = TRUE, tags = "train"),
+        ParamUty$new("etastart", tags = "train"),
+        ParamUty$new("mustart", tags = "train"),
+        ParamUty$new("start", default = NULL, tags = "train"),
+        ParamUty$new("offset", tags = "train"),
+        ParamFct$new("family",
+          default = "gaussian",
+          levels = c("gaussian", "poisson", "quasipoisson", "Gamma", "inverse.gaussian"),
+          tags = "train"),
```

Comment:
I think you are missing a parameter `link` which would get passed to a `family` object. Frustratingly this can't be passed through `...` along with controls so you will likely have to set this manually. See `?stats::family` for values `link` can take.

The simplest way to do this would to add a `family` tag to `ParamFct$new("link")` then something like `mlr3misc::invoke(get(ps$values$family), .args = ps$values$link)` then remove the `link` value before passing `ps` to `glm.fit` (I think this should work but haven't checked locally).'

Reply:
Experimentation in progress

Reply:
Got something that works. Check it out!